src/burn/engine/plan.cpp

```diff
@@ -1487,6 +1524,39 @@ extern "C" HRESULT PlanRelatedBundlesComplete(
             hr = DependencyPlanPackageComplete(&pRelatedBundle->package, pPlan);
             ExitOnFailure(hr, "Failed to complete plan dependency actions for related bundle package: %ls", pRelatedBundle->package.sczId);
         }
+
+        if (fInstallingAnyPackage && BOOTSTRAPPER_RELATION_UPGRADE == pRelatedBundle->relationType)
```

Comment:
2. As you can probably see from the first commit, I was planning on using the operation to only select MajorUpgrade bundles. But some bundles could support downgrading, so the bundle to install is actually newer and would have been labeled as Downgrade.